oracle/src/currency.rs

```diff
@@ -20,40 +20,76 @@ pub trait CurrencyInfo<Currency> {
     fn decimals(&self, id: &Currency) -> Option<u32>;
 }
 
-#[derive(Default, Debug, Clone, Eq, PartialOrd, Ord)]
-pub struct Currency {
+#[derive(Deserialize, Debug, Clone)]
+pub struct Extension {
```

Comment:
This sounds very generic, but it's actually quite specific to the fair price oracle. I think we should probably rename this, and possibly move it into the dia file